src/com/lilithsthrone/game/character/npc/NPCAI.java

```diff
@@ -0,0 +1,106 @@
+package com.lilithsthrone.game.character.npc;
+
+//import com.lilithsthrone.game.character.npc;
+
+import java.util.Random;
+
+import com.lilithsthrone.world.places.PlaceType;
+import com.lilithsthrone.game.slavery.SlavePermissionSetting;
+import com.lilithsthrone.main.Main;
+import com.lilithsthrone.utils.Vector2i;
+
+import java.util.ArrayList;
+import java.util.List;;
+
+/* This is a attempt at creating some sort of AI for NPC and try to add:
+ * 	-Roaming NPC in the map instead of static location
+ * 	-Trigger encounter upon meeting the player in the map  
+ * 	-Chase player on sight
+ * 	
+ * 
+ * 	**Time consuming idea**
+ * 	-Path finding algorithm for NPC goal(example: Slave has use_you permission could have to get to you before using you instead of teleporting)
+ * 
+ * 
+ * PS: Java is not my main language 
+ */
+
+public abstract class NPCAI {
+	public static void AIEndTurn(NPC CurrentNpc)
```

Comment:
If I understand properly it should be call NPCArtificialIntellignece?

Reply:
No, that part is defendable. You could argue the class needs to be called NpcAi or NPCAI or NpcAI or NPCArtificialIntelligence or NpcArtificialIntelligence but that's not what I'm bothered about.

All your method names start with an uppercase character; this is confusing because I expect things that start with an uppercase character (check the rest of the codebase) to be either a class, interface or enum. Things in all caps are constants. camelCase is used for variable names and method names, and I'd like to see your code adhere to the same convention for ease of reading.